app_feup/lib/view/Widgets/generic_expansion_card.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:expansion_tile_card/expansion_tile_card.dart';
+
+abstract class GenericExpansionCard extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return GenericExpansionCardState();
+  }
+
+  String getTitle();
+  Widget buildCardContent(BuildContext context);
+}
+
+class GenericExpansionCardState extends State<GenericExpansionCard> {
+  final GlobalKey<ExpansionTileCardState> cardA = GlobalKey();
+  final double padding = 12.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+        padding: EdgeInsets.symmetric(vertical: 10, horizontal: 20),
+        child: ExpansionTileCard(
+          baseColor: Color.fromARGB(0, 0, 0, 0),
```

Comment:
Avoid hard coded colors. Try
```suggestion
          baseColor: Theme.of(context).backgroundColor,
```
or remove the property

Reply:
I would just remove the property, or even better, pass it as an optional parameter to the constructor to be easily used elsewhere if needed.

Reply:
If I remove it, the base color will be the background color, so there won't be any visible changes.
Allowing the selection of the card's color with an optional parameter would be a great "feature", but as it won't be used in these (specific) cards, I think we shouldn't implement it now. We could do it in a future situation, when we'll actually need coloured expandable cards. And, in that case, the 'baseColor' property will be important, so in my opinion it should already stay in this code.
What do you think?

Reply:
I don't think I understood what you're suggesting. Keep the property and make it adjust to the theme?

Reply:
Yes, keep the property with the value of the background color. 
Although it's not really useful now, I think it should stay because it can be important to, if necessary, develop the possibility for the card to have its own color.

Reply:
If the card is supposed to not have its own color for now, then the property should not be specified. There is no guarantee that we'll use the `backgroundColor` in the future for the page background's color and doing so may lead to issues later on. It's also simple to add the color via parameter later on, so that's fine

Reply:
Ok then, it makes sense. I'm removing it